overlord/snapstate/handlers_components.go

```diff
@@ -342,8 +342,15 @@ func (m *SnapManager) doMountComponent(t *state.Task, _ *tomb.Tomb) (err error)
 	return nil
 }
 
+// ReadComponentInfo reads the snap's component and returns a ComponentInfo.
+func ReadComponentInfo(snapInfo *snap.Info, csi *snap.ComponentSideInfo) (*snap.ComponentInfo, error) {
+	compName, compRev := csi.Component.ComponentName, csi.Revision
+	mountDir := snap.ComponentMountDir(compName, compRev, snapInfo.InstanceName())
+	return readComponentInfoAt(mountDir, snapInfo, csi)
+}
+
 // Maybe we will need flags as in readInfo
-var readComponentInfo = func(compMntDir string, snapInfo *snap.Info, csi *snap.ComponentSideInfo) (*snap.ComponentInfo, error) {
+var readComponentInfoAt = func(compMntDir string, snapInfo *snap.Info, csi *snap.ComponentSideInfo) (*snap.ComponentInfo, error) {
```

Comment:
Can you explain the At suffix please?

Reply:
I needed to distinguish this from the other higher level helper and since this takes a path, `...At` seemed appropriate. The tests could be refactored to just mock the new helper instead but it's a somewhat large and unrelated change in this PR (maybe as a follow-up)